### PR TITLE
Generate index.json

### PR DESCRIPTION
**Pull request by mtojek**
Issue: https://github.com/elastic/package-registry/issues/469

_Let's first merge this PR: https://github.com/elastic/package-registry/pull/468_

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+type indexData struct {
+	ServiceName string `json:"service.name"`
+	Version     string `json:"version"`
+}
+
+func indexHandler(cacheTime time.Duration) (func(w http.ResponseWriter, r *http.Request), error) {
+	data := indexData{
+		ServiceName: "package-registry",
+		Version:     "0.4.0",
```
Comment (ruflin) [inline index.go:21]: Nit: Could have this as a constant in the main.go file?

File: index.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+type indexData struct {
+	ServiceName string `json:"service.name"`
+	Version     string `json:"version"`
+}
+
+func indexHandler(cacheTime time.Duration) (func(w http.ResponseWriter, r *http.Request), error) {
+	data := indexData{
+		ServiceName: "package-registry",
```
Comment (ruflin) [inline index.go:20]: Nit: Could we have this as a constant?

**Comment by mtojek** [comment]
> Changelog?

Yup, already added.

File: index.go (hunk 1 above)
Comment (mtojek) [reply index.go]: Fixed

File: index.go (hunk 2 above)
Comment (mtojek) [reply index.go]: Fixed

File: main.go [hunk 3]
```diff
@@ -121,13 +124,20 @@ func getRouter(config Config, packagesBasePath string) (*mux.Router, error) {
...
 
 	router := mux.NewRouter().StrictSlash(true)
+	router.HandleFunc("/", indexHandlerFunc)
+	router.HandleFunc("/index.json", indexHandlerFunc)
 	router.HandleFunc("/search", searchHandler(packagesBasePath, config.CacheTimeSearch))
 	router.HandleFunc("/categories", categoriesHandler(packagesBasePath, config.CacheTimeCategories))
 	router.HandleFunc("/health", healthHandler)
 	router.HandleFunc("/favicon.ico", faviconHandleFunc)
-	router.PathPrefix("/").HandlerFunc(catchAll(http.Dir(config.PublicDir), config.CacheTimeCatchAll))
+	router.PathPrefix("/epr").HandlerFunc(catchAll(http.Dir(config.PublicDir), config.CacheTimeCatchAll))
```
[the remark this answers is not in the document]
Comment (mtojek) [reply main.go:139]: From the user perspective - no, they're already exposed. 

Speaking about the implementation, this has been already adjusted (see the master branch). Archives exposed via the `/epr` endpoint are built dynamically, not served from the file system.

**Comment by mtojek** [comment]
> I would prefer to keep it in index.json so other tools can read it from the filesystem. This approach requires an HTTP request if I understand correctly.

This PR is only for the main `index.json` file. I've just started working on the package `index.json` files (PR soon).

**Comment by jfsiii** [comment]
> This PR is only for the main index.json file.

Yes, that was my understanding.

I'll try to keep an eye out, but can you add me on that PR when you submit it? Thanks

**Pull request merged**